text/3594-expose-stackrealign-attribute.md

```diff
@@ -0,0 +1,85 @@
+- Feature Name: (`realign-stack-attr`)
+- Start Date: (2024-03-26)
+- RFC PR: [rust-lang/rfcs#3594](https://github.com/rust-lang/rfcs/pull/3594)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Expose the `alignstack` function attribute to rustc.
+
+# Motivation
+[motivation]: #motivation
+This is usefull when you have no guarantees about the alignment of the stack in an extern "C" function.
+Also verbatim from the attribute reference:
+Legacy x86 code uses 4-byte stack alignment. Newer aligned SSE instructions (like ‘movaps’) that work with the stack require operands to be 16-byte aligned. This attribute realigns the stack in the function prologue to make sure the stack can be used with SSE instructions.

```

Comment:
Something can't be right here. We can already write code that uses SSE instructions, on stable Rust today, without this attribute. So either the attribute is not needed here, or there's a critical soundness bug.

Reply:
the question is whether rustc forecfully aligns functions that use SSE or not.

Reply:
I think it does. And if it doesn't then it needs to be fixed to do so. Have you actually tried this?

Under no circumstances will we expect Rust programmers to write such an attribute just so that they can use SSE instructions.

Reply:
removed that paragraph